conda_lock/lookup.py

```diff
@@ -3,27 +3,42 @@
 import requests
 import yaml
 
+from typing_extensions import TypedDict
+
+
+class MappingEntry(TypedDict):
+    conda_name: str
+    # legacy field, generally not used by anything anymore
+    conda_forge: str
+    pypi_name: str
+
 
 class _LookupLoader:
     def __init__(self) -> None:
         self._mapping_url = "https://raw.githubusercontent.com/regro/cf-graph-countyfair/master/mappings/pypi/grayskull_pypi_mapping.yaml"
-        self._pypi_lookup: Optional[dict] = None
-        self._conda_lookup: Optional[dict] = None
+        self._pypi_lookup: Optional[Dict[str, MappingEntry]] = None
+        self._conda_lookup: Optional[Dict[str, MappingEntry]] = None
 
     def resolve(self) -> None:
         res = requests.get(self._mapping_url)
         res.raise_for_status()
         self._pypi_lookup = yaml.safe_load(res.content)
+        # lowercase the pypi names
+        assert self._pypi_lookup is not None
+        self._pypi_lookup = {k.lower(): v for k, v in self._pypi_lookup.items()}
+        for v in self._pypi_lookup.values():
+            v["pypi_name"] = v["pypi_name"].lower()
```

Comment:
perhaps also normalize `_` to `-`?

Reply:
yeah that feels prudent

Reply:
Here's "the law": https://peps.python.org/pep-0503/#normalized-names

```
import re

def normalize(name):
    return re.sub(r"[-_.]+", "-", name).lower()
```